src/ccall.cpp

```diff
@@ -1490,13 +1491,15 @@ static jl_cgval_t emit_ccall(jl_codectx_t &ctx, jl_value_t **args, size_t nargs)
     };
 #define is_libjulia_func(name) _is_libjulia_func((uintptr_t)&(name), #name)
 
+    static jl_ptls_t (*ptls_getter)(void);
+    // directly accessing the address of an ifunc can cause compile-time linker issues
+    // on some configurations (e.g. AArch64 + -Bsymbolic-functions), so we guard the
+    // `&jl_get_ptls_states` within this `#ifdef` guard, and use a more roundabout
+    // method involving `jl_dlsym()` on Linux platforms instead.
 #ifdef _OS_LINUX_
```

Comment:
This actually makes the `static` useless. The performance isn't super critical here so if you want to just get rid of the `static` that's fine. Otherwise you can write it as.

```c++
static auto ptls_getter = [] {
#ifdef _OS_LINUX_
    jl_ptls_t (*p)(void);
    jl_dlsym(....);
    return p;
#else
    return jl_get_ptls_states;
#end
}();
```

Reply:
Wow, that's available even all the way back to `GCC 4.8.5`.  I'm amazed.  And yes, you're right this completely defeats `static`, but I like this fancy lambda initialization.  I'm gonna go with that since compiler support is good enough.

Reply:
This is standard C++11 feature and we require C++11 so that shouldn't be very surprising......